event/Event.proto

```diff
@@ -0,0 +1,79 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+syntax = "proto3";
+
+package skywalking.v3;
+
+option java_multiple_files = true;
+option java_package = "org.apache.skywalking.apm.network.event.v3";
+option csharp_namespace = "SkyWalking.NetworkProtocol.V3";
+option go_package = "skywalking/network/event/v3";
+
+import "common/Common.proto";
+
+service EventService {
+  // When reporting an event, you typically call the collect function twice, one for starting of the event and the other one for ending of the event, with the same UUID.
+  // There are also cases where you have both start time and end time already, for example, when exporting events from a 3rd-party system,
+  // the start time and end time are already known so that you can call the collect function only once.
+  rpc collect (stream Event) returns (Commands) {
+  }
+}
+
+message Event {
+  // Unique ID of the event. Because an event may span a long period of time, the UUID is necessary to associate the
+  // start time with the end time of the same event.
+  string uuid = 1;
+
+  // The source object that the event occurs on.
+  Source source = 2;
+
+  // The name of the event. For example, `Reboot`, `Upgrade` etc.
+  string name = 3;
+
+  // The type of the event. This field is friendly for UI visualization, where events of type `Normal` are considered as normal operations,
+  // while `Error` is considered as unexpected operations, such as `Crash` events, therefore we can mark them with different colors to be easier identified.
+  Type type = 4;
+
+  // The detail of the event that describes why this event happened. This should be a one-line message that briefly describes why the event is reported.
+  // Examples of an `Upgrade` event may be something like `Upgrade from ${from_version} to ${to_version}`.
+  // It's NOT encouraged to include the detailed logs of this event, such as the exception stack trace.
+  string message = 5;
+
+  // The parameters in the `message` field.
+  map<string, string> parameters = 6;
+
+  // The start time (in milliseconds) of the event, measured between the current time and midnight, January 1, 1970 UTC.
+  // This field is mandatory when an event occurs.
+  int64 startTime = 7;
+
+  // The end time (in milliseconds) of the event. , measured between the current time and midnight, January 1, 1970 UTC.
+  // This field may be empty if the event has not stopped yet, otherwise it should be a valid timestamp after `startTime`.
+  int64 endTime = 8;
+}
+
+enum Type {
+  Normal = 0;
+  Error = 1;
+}
+
+message Source {
+  string service = 1;
+  string serviceInstance = 2;
+  string endpoint = 3;
```

Comment:
We need comments to indicate how to use this, such as for a service, `instance` and `endpoint` should be empty, correct?

Reply:
Wouldn't you just use the ones you're able to specify? For example, you could fill in `service` and `endpoint` to mean an endpoint for any service instance. Or you could fill in `serviceInstance` and `endpoint` to mean the endpoint on a specific service instance.

My thinking at least. Allows you to specify something as precise as you want and gives us fields that should only be used to mean a certain thing (service should only be service ids).

Reply:
> Or you could fill in serviceInstance and endpoint to mean the endpoint on a specific service instance.

This is one thing I want to highlight. We don't have `endpoint in instance` for example. We only have 
- Service
- Instance of Service
- Endpoint of Service

Reply:
The usage in my mind is, if the event occurs on a service ONLY, the serviceInstance and endpoint are optional, if it occurs on serviceInstance, the service is mandatory and endpoint is optional, if it occurs on an endpoint, `service` and `endpoint` are mandatory and `serviceInstance` is optional (depends on whether the event occurs a particular service instance or all instances of a service), does it make sense?

Reply:
Misunderstanding on my part about instance and endpoint but I agree with @kezhenxu94.

Reply:
Sure I agree with those, I just want them in the proto.

Reply:
> Sure I agree with those, I just want them in the proto.

Comments are added